\documentclass[10pt]{article}
\usepackage{a4wide}
\usepackage[german,english]{babel}
\usepackage[latin1]{inputenc}
\usepackage{url}
\usepackage{epsfig}
\usepackage[super]{nth}

%\usepackage{times} % okay (KURZ)
%\usepackage{charter} % okay (MITTEL)
\usepackage{palatino} % okay (LANG)

\parindent0em
\parskip1ex

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% MACROS AND DEFINITIONS
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newcommand{\nobox}{\setlength{\unitlength}{.8mm}%
                      \begin{picture}(10,2)
                      \end{picture}}
\newcommand{\boxone}{\setlength{\unitlength}{.8mm}%
                     \begin{picture}(10,2)
                       \put(0,0){\rule{8mm}{1.6mm}}
                     \end{picture}}
\newcommand{\boxtwo}{\setlength{\unitlength}{.8mm}%
                     \begin{picture}(10,2)
                       \put(0,0.0){\line(1,0){10}}
                       \put(0,0.5){\line(1,0){10}}
                       \put(0,1.0){\line(1,0){10}}
                       \put(0,1.5){\line(1,0){10}}
                       \put(0,2){\line(1,0){10}}
                       \put(0,0){\line(0,1){2}}
                       \put(10,0){\line(0,1){2}}
                     \end{picture}}
\newcommand{\boxthree}{\setlength{\unitlength}{.8mm}%
                     \begin{picture}(10,2)
                       \put(0,0){\line(1,0){10}}
                       \put(0,0){\line(0,1){2}}
                       \put(0,2){\line(1,0){10}}
                       \put(10,0){\line(0,1){2}}
                       \multiput(0,0)(1.7,0){4}{\line(5,2){4.8}}
                       \multiput(4.8,0)(1.7,0){4}{\line(-5,2){4.8}}
                     \end{picture}}

\newcommand{\ignore}[1]{}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% TITELSEITE
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}

\thispagestyle{empty}
\vspace*{20ex}
\begin{center}
\large
{\bf Antrag an die Deutsche Forschungsgemeinschaft}\\
Kennedyallee 40, 53175 Bonn\\[5ex]
auf Gewährung einer Sachbeihilfe\\[10ex]
{\bf Dual Sparse De-Bruijn Subgraphs (DSDS)\\
for High-throughput Sequencing}\\[5ex]
Prof.\ Dr.\ Jens Stoye\\
Arbeitsgruppe Genominformatik\\[2ex]
Technische Fakultät der Universität Bielefeld\\[2ex]
und\\[2ex]
Institut für Bioinformatik\\
Centrum für Biotechnologie der Universität Bielefeld
\end{center}
\newpage
\thispagestyle{empty}~
\newpage
\setcounter{page}{1}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% ALLGEMEINE ANGABEN
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{General information (Allgemeine Angaben)}

Proposal for a research grant (new application).

\subsection{Applicant (Antragsteller)}

\setlength{\tabcolsep}{0em}

\begin{tabular}{l@{~~~~~~}l}
Name:              & Prof.\ Dr.\ Jens Stoye\\
Employment status: & Universitätsprofessor (C4)\\
Date of birth:     & 17.\ 03.\ 1970\\
Nationality:       & german\\
Institution:       & Technische Fakultät und Institut für Bioinformatik\\
                   & der Universität Bielefeld\\
Office address:    & Universität Bielefeld\\
                   & Technische Fakultät\\
                   & AG Genominformatik\\
                   & 33594 Bielefeld\\
Telephon:          & 0521 - 106 - 3852\\
Telefax:           & 0521 - 106 - 6495\\
E-Mail:            & stoye@techfak.uni-bielefeld.de\\
Private address:   & Kesselstraße 3\\
                   & 33602 Bielefeld\\
                   & 0521 -- 9675479
\end{tabular}

See attached documents for a tabular CV.

\subsection{Topic (Thema)}

{\bf TODO Jens: reduce length by a few characters}

Watson-Crick complement invariant version of sparse De-Bruijn subgraphs for
analysis and assembly of high-throughput sequencing data.

%% 106 characters
%% Max allowed: 104 characters

Watson-Crick-Komplement-unabhängige Version von sparse De-Bruijn-Teilgraphen
für die Analyse und Assemblierung von Hochdurchsatz-Sequenzierdaten.

%% 126 characters
%% Max allowed: 104 characters

\subsection{Scientific discipline and field of work (Fachgebiet und
Arbeitsrichtung)}

Scientific discipline: Computer Science, algorithms engineering\\
Field of work: Sequence analysis

\subsection{Scheduled total duration (Voraussichtliche Gesamtdauer)}

Duration: 30 months

\subsection{Application period (Antragszeitraum)}

First proposal: Yes\\
Time period: \nth{1} January 2008 -- \nth{30} June 2010\\
Funding commence: \nth{1} January 2008

\subsection{Summary (Zusammenfassung)}

High-throughput DNA sequencing methods in use today are able to produce
staggering amounts of data on a daily basis that demand extensive computing
resources to assemble, finish and process genomic information. The Center for
Biotechnology (CeBiTec) at Bielefeld University will soon have its own
high-throughput sequencing machine, a patented technology developed by 454 Life
Sciences, which is capable of producing millions of raw DNA bases per hour.
However, what makes sequencing a challenge today is the subsequent computational
problem of correctly assembling the reads into the original sequence is .

This project aims to develop a novel approach to genome assembly based on Dual
Sparse De-Bruijn Subgraphs (DSDS). De-Brujin subgraphs are already being studied
by members of our research group. The new approach consists of a version of our
current model that exploits the Watson-Crick complementarity nature of the DNA.
The project consists of two parts. First, we will study and develop the new
graph, and design the necessary data structures for efficient implementation. In
the second part of the project, will apply our new approach in practice to
genomic assembly of DNA data sequenced at the CeBiTec institute in Bielefeld.

{\bf TODO Jens: German version}

%% 14 lines, 1047 characters
%% Max allowed: 15 lines, 1600 characters

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% STAND DER FORSCHUNG, EIGENE VORARBEITEN
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{State of the art, preliminary work (Stand der Forschung, eigene
Vorarbeiten)}

\subsection{State of the art (Stand der Forschung)}

De Bruijn graphs were first studied in the end of the \nth{19} century,
although in an implicit form, and were formally defined in 1946 by N.~G.~de~
Bruijn~\cite[Chapter 3]{moreno}. They are directed graphs with a simple and
clear definition that are easy to build, and that have interesting properties:
they are regular, have small diameter, and are both Hamiltonian and Eulerian. As
a result, de Bruijn graphs have been used in diverse applications such as
network models, pseudo-random number generation, and DNA analysis~\cite{moreno}.
Similar graphs have also been used for the identification of repeat
families~\cite{raphael04}, but the first use in Bioinformatics was probably the
Eulerian path approach to sequence assembly proposed by Idury and
Waterman~\cite{idury95} and extended by Pevzner, Tang and
Waterman~\cite{pevzner01a}.

Despite the success achieved by the sequence assembler developed by Pevzner and
co-workers for bacterial genomes, de Bruijn graphs have not been further
explored in computational biology. In some works they appear slightly modified,
and are more often used as a multiple alignment displayer than a basic
tool~\cite{pevzner04,raphael04}. As Myers~\cite{myers05} points out, the main
problem with de Bruijn graphs is that, in the original definition, they are
space inefficient.

A $d$-dimensional de Bruijn Graph $G=(V,A)$ on an alphabet $\Sigma$ is a
directed graph defined as follows:
\begin{eqnarray*}
   V &=& \Sigma^d \\ 
   A &=& \lbrace (u,v)\ |\ u,v \in V \mbox{ and }
   u_{i+1} = v_i, \mbox{ for all } 1 \leq i < d \rbrace,
\end{eqnarray*}
%%
where $u_i$ is the $i$-th character of the string represented by $u$. Strings of
length at least $d$ over the same alphabet describe walks on the $d$-dimensional
de Bruijn graph. Given a set of strings, we define the associated
$d$-dimensional \emph{de Bruijn subgraph} as the subgraph of the $d$-dimensional
de Bruijn graph that contains all the walks described by these strings but no
extra vertex or arc.

Sequence-associated de Bruijn subgraphs have a nice asymptotic behavior: The
maximum number of nodes increases linearly with the size of the input and
decreases with the dimension $d$ of the graph. The main problem is that,
although these graphs scale well with the string set size, graphs corresponding
to even relatively small genomes (such as of bacteria) are already prohibitively
large for an average computer. This could probably explain why de Bruijn graphs
have not been further used in Bioinformatics.

Another disadvantage of simple de Bruijn graphs for genome assembly is that they
require cutting the sequence reads into small pieces to finally build the graph.
Myers~\cite{myers05} have suggested that this step might not be necessary. In
fact, we have successfully constructed a hybrid structure based on de Bruijn
subgraphs without explicitly using sequence comparisons, with the added
advantage of allowing the representation of a whole genome of a higher-order
organism in the memory of a typical personal computer.

In a typical sequence-associated de Bruijn subgraph, the graph branches are more
important than the nodes in non-branching paths, and most of the nodes in such a
graph are exactly in non-branching paths. Hence, the first step to reduce its
size is to contract non-branching paths into single nodes containing sequences
of unlimited length, similarly to what Idury and Waterman~\cite{idury95} have
done in their assembly algorithm.

In order to avoid constructing the large non-contracted graph, we define an
indexed structure, called \emph{sequence graph}, formed by a relaxed definition
of a de Bruijn graph and a mapping from $d$-tuples to the nodes that represent
them. In the relaxed de Bruijn graph, the length of a sequence in a node is
unbounded. This allows us to create a compact representation of sparse de Bruijn
subgraphs without non-branching paths, while the access to a particular
$d$-tuple is still possible via the index. Our initial experiments show that the
sequence graph is able to store the same information as a de Bruijn subgraph
using roughly $25\%$ of the memory. We also notice a proportional reduction of
the time spent in constructing the graph.

We believe that the sequence graph can be further improved by exploiting a
peculiarity of genomic data, namely, the reverse-complementarity nature of DNA.
A DNA molecule is formed by two complementary strands that bind according to
pairing rules in the double-helix structure discovered by Watson and Crick,
namely guanine (G) pairs with cytosine (C), and adenine (A) pairs with thymine
(T). Current sequencing methods are not able to distinguish from which strand a
given sequence comes and, as a result, each sequence read is represented twice
in sequence graph. Our aim is to avoid this duplication by adding the concept of
complementarity both in the index and in the relaxed de Bruijn graph. The
structure we propose is called Dual Sparse De-Bruijn Subgraph. In case such a
structure can be efficiently built and updated, we will be able to double the
efficiency of the structure we already have both in terms of memory requirements
and construction time.

\subsection{Preliminary work, progress report (Eigene Vorarbeiten,
Arbeitsbericht)}

The proposed data structure has several applications in genome
analysis, which is the main subject of most of our group activities,
as can be seen below.

%% Begin ignore block
\ignore{
\begin{enumerate}
\item Sequence searching
\item Genome evolution
\item Primer design / microarray design
\item Repeat analysis, other than primer design (write 1 paragraph, cite
      Reputer paper)
\end{enumerate}
}
%% End ignore block

\paragraph{Sequence Searching.}
The group worked in close collaboration with the Research Group for
Practical Computer Sciences on indexing methods for large amounts of
genomic data. Among them are the \emph{suffix
  trees}~\cite{GIE-KUR-STO-2003}, which are at moment very popular in
the field of Bioinformatics.  Suffix trees can be used in many
different applications~\cite{GUS-1997}. The identification of repeated
sequences~\cite{KUR-CHO-OHL-SCH-STO-GIE-2001,STO-GUS-2002,GUS-STO-2004}
is one of them.

An \emph{affix tree}~\cite{STO-2000} is a generalization of suffix
trees for applications related to the search for a given pattern in a
text. They represent a connection between suffix trees and their
coresponding complementary reverse \emph{prefix trees}.  Each
substring of a text is therefore represented twice (as if we read it
from both left to right and right to left). This gives support to
flexible search strategies. The linear memory requirement of affix
trees is known since its development in Bielefeld in 1995.  An
efficient algorithm for constructing them was presented later by
Maa\ss~\cite{MAS-2003}. At the moment we are exploring an application
to the search of special non-homogeneous patterns, where the use of
affix trees can improve considerably the search efficiency.

While suffix trees, and specially affix trees, theoretically allow the
implementation of both time and space efficient algorithms, the amount
of memory required by such structures in praxis are not negligible. An
efficient alternative is the use of \emph{suffix arrays}.  They can be
used in almost all scenarios where suffix trees can be applied, and
their representation requires much less memory in
praxis~\cite{ABO-KUR-OHL-2004}. An algorithm for the construction of
suffix arrays from genomic data was developed in the Research Group
Genome Informatics~\cite{SCH-STO-2005}. The algorithm may be used in
many different applications, and has a running-time comparable to the
existent algorithms, being even faster in many cases.

An even less memory consuming index for database searches is the so
called {\em $q$-gram hash}, where the positions of every $q$-grams
(substrings of length $q$) are stored in a table. Based on this index,
we were able to design a filter algorithm in collaboration with Gene
Myers (Janelia Farms Research Campus of the Howard Hughes Medical
Institute). Besides the filter algorithm, the database search software
tool SWIFT~\cite{RAS-STO-MYE-2006} was also a result of this
collaboration. SWIFT not only requires much less memory than similar
software tools, but is also 25 times faster than the most popular of
them, BLAST, without any lost of sensibility.

\paragraph{Genome evolution modeling.} Information about the evolution 
of genomes may be obtained from the global structure of the gene
arragements in them. In this case, the corresponding maximum parsimony
assumption, namely the minimum number of transformations (inversions,
translocations, transpositions, fusions, and fissions of genome
pieces), is used to transform a genome into another one. The related
mathematical theory was studied by Hannenhalli and
Pevzner~\cite{HAN-PEV-1995B,HAN-PEV-1995C,HAN-1996} during the 1990's.
The highly technical content of their work was at that time a big
barrier to the area. Together with Anne Bergeron (Montreal), we could
make important steps to the simplification of this
theory~\cite{BER-HEB-STO-2002,BER-MIX-STO-2004,BER-MIX-STO-2005,BER-STO-2003B}.
Errors in the original works could also be identified and corrected
during the simplification process~\cite{BER-MIX-STO-2005B}. Another
object studied by our group are gene clusters, which are sets of genes
that appear close to each other in different, not so closely related,
genomes.  In~\cite{HEB-STO-2001,HEB-STO-2001B,SCH-STO-2004} we find
many different algorithms able to efficiently identify gene clusters
in several genomes. At the moment, this work is being extended and
applied to real data. The results are also being combined to data from
other sources, like gene expression studies.


%% Begin ignore block
\ignore{
\paragraph{Computer science methods for mass-spectrometry.} 

After the human genome sequencing project, the biotechnology focus
moves from sequencing a species to sequencing of single exemplars, and
from the genome analysis to the analysis of the metabolome and
proteome of organisms. Mass-spectrometry, the most used analysis in
these cases, may be applied in the re-sequencing, SNP identification,
and forensic medicine. Methods for the mass spectrometer calibration,
statistical based methods for protein identification,
DNA-mass-spectrometry, as well as mass-spectrometry related
combinatorial problems were developed at Mr. B{\" o}cker's junior
research group.

To be more precise, in the group we develop methods, which allow the
comparison between two mass-spectra. Specially alignments between real
and theorethi\-cally-predicted ones. This makes an efficient and
robust protein identification possible. One of the methods developed
by the group allows the calculation of one of these alignments
significance ($p$-value). In addition, the group works in methods for
the analyses of mass-spectrometry rough-data (peak detection): these
data contain many kinds of noise, and statistic methods may improve
also this step of the analysis.

The analysis of DNA with the help of base-specific cleavage and
mass-spectro\-metry is a new method for the study of individual
genomes.  The biochemistry behind the method is being optimized at the
moment, and the first results are promising. This area contains lots
of computer science problems, which can be investigate with
success~\cite{BOE-2003,BOE-2004}, also in common subjects of the
DNA/MS-analysis, like the search for SNPs~\cite{BOE-2003B,
  HAR-STO-BOE-ROD-HIL-JUR-BOO-2003,
  STA-ZAB-MEE-REM-GAN-STO-HAR-RON-ROD-BOE-BOO-2004, EHR-BOE-BOO-2005,
  BOE-2006} or pathogen
identification~\cite{LEF-HON-BOE-STO-WIN-SCH-MOT-BOO-2004}.

The computer science basis of every mass-spectrometric biomolecular
experiment are weighted strings: a biomolecule is a sequence of amino
acids or nucleotides, and can therefore be represented as strings.
Each of these building-blocks has a known weight, and the weight of
the entire molecule is the sum of the weights of each single
building-block. If a molecule of mass $m$ is observed in
mass-spectrometry experiment, many related questions arise
automatically, like: Is there any string with mass $m$? Which strings
have mass $m$? A statistical analysis of mass-spectrometry data is
only possible after these questions are answered. And we approached
these and related questions in our
group~\cite{CIE-ERL-LIP-STO-WEL-2001, BOE-ZSU-2005, BOE-ZSU-2005B}.
}
%% End ignore block

\paragraph{Primer design / microarray design.} 
Systems biology explores biological systems and their interactions
both mathematically and algorithmically. 
%% Begin ignore block
\ignore{
Heterogeneous data from
different sources, like DNA-microarrays, protein mass-spectrometry,
protein interaction, and transcription factor biding ideally give a
consistent overview, from which it is possible to determine the
architecture (e.g. the network topology) and other necessary system
parameters.
}
%% End ignore block
We aim at the development of methods for the improvement of data
quality, which ideally could be applied just after or even during the
data collection, as well as methods for the integration of data from
different experiments. In collaboration with the chair of Genetics, we
analyze the regulatory network of {\it Corynebacterium glutamicum}.
The experience shows that it is difficult to infer a regulatory
network from large scale expression data. This is partially due to the
not negligible noise factor in microarray results.  Therefore we focus
on an early step of the data production pipeline, and optimize the
microarray design through a careful choice and positioning of the
millions of probes in the chip~\cite{RAH-2003,RAH-2003B,RAH-GRA-2004}.
The suffix arrays mentioned at the beginning of this text find also
here an application.

%% Begin ignore block
\ignore{
In some of the applications, it is impossible to avoid that several
different target genes hybridise with the same olignucleotide probe.
Alternative splicing analysis and microorganism detection are examples
of such experiments. During the design of these DNA-chips we may need
the help of combinatorial optimization methods, like integer linear
programming~\cite{KLA-RAH-SCH-2004}. In order to decode the performed
experiments, we use methods from the statistical group
testing~\cite{SCH-TOR-RAH-2003}.

Although expression data provide valuable information, this data alone
is not enough to infer the complete network. However, it is always
possible to use hints provided by other sources of data, like DNA
binding domain signals in proteins and transcription factor binding
sites at the gene regulated upstream region. Because upstream
regulatory regions provide only weak signals, the analysis of these
regions require a precise statistical
analysis~\cite{DIE-RAH-VIN-2004,RAH-MUE-VIN-2003}.

In later steps we use stochastic methods, like simulated annealing, in
order to choose the most plausible between many concurrent models for
a regulatory network.
}
%%End ignore block

\paragraph{Repeat analysis.}

Identifying repetitive elements in the genome of eukaryotes is an
important task both if one is interested in study them and if one
needs to avoid them. Although the sequence of some repetitive elements
may be characterized, which allows repeat identification by
traditional sequence comparison methods, the experience shows that
most of the eukaryotes have very specific families of repetitive
elements. For completely sequenced genomes, the {\em de novo}
identification of such sequences may be done using suffix trees. A
software tool called REPuter~\cite{KUR-CHO-OHL-SCH-STO-GIE-2001} was
developed based on this approach. In a collaboration with Prof.\ Bernd
Weisshaar (Chair of Genome Research, Universität Bielefeld), we are
developing a De Bruijn subgraph based method for doing the same with
collections of reads of an incompletely sequenced genome. Also here
the size of de Bruijn subgraphs is a considerable barrier to the
development of software tools for personal computers.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% ZIELE UND ARBEITSPROGRAMM
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Goals and work schedule (Ziele und Arbeitsprogramm)}

\subsection{Goals (Ziele)}

\begin{figure}
  \epsfxsize=0.6\textwidth
  \begin{center}\epsfbox{graphics}
  \end{center}
  \caption{Memory usage (in gigabytes) for de Bruijn subgraphs and their correponding sequence graphs. The number of sequences corresponds to the number of 700 bases length randomly taken subsequences of an {\em Arabidopsis thaliana} chromosome.}
  \label{fig.graph}
\end{figure}

\begin{figure}
  \epsfxsize=\textwidth
  \begin{center}\epsfbox{advantages}
  \end{center}
  \caption{Four graphs for assembling the sequences {\tt AAACCC} and {\tt
      TTGTGGG}. }
  \label{fig.advantages}
\end{figure}

{\bf TODO Z\'e: highlight goals (A, B, C, D) -- see work schedule}

While working with a de Bruijn based approach to repeat finding, we
were able to develop an algorithm for building the sequence graph, the
compact version of a de Bruijn subgraph, directly from the given set
of sequences. The graph on Figure~\ref{fig.graph} presents the amount
of memory used by de Bruijn graphs and by the corresponding sequence
graphs. Figure~\ref{fig.advantages} exemplifies how memory can be
saved: The de Bruijn subgraph containing both the sequences {\tt
  AAACCC} and {\tt TTGTGGG}, with the respective complements
(Fig.~\ref{fig.advantages}.2), contains non-branching paths. Since the
information we are interested in are exactly in the graph branches,
the same information represented by the 16 nodes, 48 characters, and
14 edges (Fig.~\ref{fig.advantages}.2) can be represented by 2
nodes, 20 characters and no edge at all (Fig.~\ref{fig.advantages}.3).

Most of the Bioinformatics applications need to deal both with the
input set and its reverse-complement. Without considering the
reverse-complement, important information can be lost, as
Figure~\ref{fig.advantages}.1 shows. In this case, because only the
input set is considered, one may conclude that the given sequences
belong to independent DNA sequences, whereas
Figure~\ref{fig.advantages}.2 shows that both sequence are part of
different strands of the same DNA molecule. The direct consequence of
this is that in a de Bruijn graph for the DNA alphabet, the
information is usually represented twice.

The main goal of this project is to take advantage of the DNA sequence
complementarity to reduce even more the graph size. In the example
shown in Figure~\ref{fig.advantages}.4, the reduction achieved is of
50\%. Of course this is a very special case. In the real life, we will
be facing many problems not shown in the example, like
self-complementary sequences.

The first goal of this project is therefore dedicated to the
investigation of the proposed graph. After understanding the its
characteristics, we will implement a genome assembler so that we can
compare the performance of the new graph with previously existing
applications, which consists of our second goal.

We choose the genome assembly problem for mainly two reasons:

\begin{itemize}
\item The de Bruijn graph approach do genome assembly is well studied
  and documented. Not only the assembly, but also a sequencing error
  correction method are described in the literature.
\item From the known applications for de Bruijn graphs, the genome
  assembly is the only one with which the group has no previous
  experience. Since the application is well documented, it is a good
  chance to expand our group's knowledge without risking delaying the
  project.
\end{itemize}

The third goal of the project is the validation of the assembler. For
attaining this goal, we count with the collaborations both in
Bielefeld and in Brazil.

The above mentioned goals form the project's main stream.  In parallel
to this main stream we have a fourth goal, which is the adaptation of
the repeat detection method being developed in our group to use DSDSs.
At the end, the project must produce an implementation of DSDSs and two
validated applications using them.

\subsection{Work schedule (Arbeitsprogramm)}

The project's work schedule consists of four phases, as follows.

\subsubsection*{(A) Design}

The first phase of the project consists of understanding the properties of the
Dual Sparse De-Bruijn Subgraphs (DSDS) and designing the necessary data
structures for implementing the genome assembler (phase B) and integrating it
with our existing repeat detection functions (phase D).

{\bf TODO Z\'e: List functions that the DSDS should support}

In this phase, the storage requirements for the data structure will be studied
in detail, so that the memory requirements for the genome assembler
and repeat detection functions can be predicted based on the size of the input
(sequencing data).

The implementation phase (phase B) begins three months after the beginning of
the design phase. Both phases will continue in parallel as the implementation is
likely to require changes in the data structue design. The design of basic data
structures are expected to reach a stable state by October 2008 and a final
version should be ready by the end of the year, although the first quarter of
the second year may be also used in case any serious design problems arise
during the implementation phase.

\subsubsection*{(B) Implementation}

In this phase, the necessary data structures will be implemented in Java by a
team of two student programmers lead by a post-doctorand researcher. An initial
working implementation providing the basic functions should be ready by October
2008 so that the DSDS design can be exhaustively evaluated and validated until
the end of the first year together with a final implementation of the DSDS.

The implementation of a genome assembler as a stand-alone software package then
starts in early 2009. The final version of the genome assembler should be ready
by the end of 2009.

\subsubsection*{(C) Integration of existing repeat detection algorithm with DSDS}

Once the implementation of the DSDS becomes stable, it will be integrated with
the existing repeat detection algorithms already implemented at the Genome
Informatics research group.

\subsubsection*{(D) Evaluation of genome assembler}

The genome assembler built in phase B will be used with real data to assess
the results and the success of the DSDS approach. This evaluation will consist
of assembly sequencing data generated by the Center for Biotechnology (CeBiTec)
in Bielefeld and our partners in Brazil. The validation of the results will
involve biologists from the Genome Research group of Prof.\ Bernd Weisshaar and
from the group of Dr. Felipe Rodrigues da Silva at Embrapa, in Brazil. Running
times and memory requirements will also be measured in order to compare the
results with those obtained with similar software packages.

The evaluation phase will start as soon as the first implementation of the
basic DSDS functions is ready, and it will last until the end of the project,
in parallel with the implementation (phase B) and integration (phase D) phases.

\subsubsection*{Summary of work schedule}

\begin{center}
\setlength{\tabcolsep}{0cm}
\begin{tabular}{|@{~~}c@{~~}l@{~~}|p{3.2cm}|p{3.2cm}|p{3.2cm}|}
\hline
\multicolumn{2}{|c|}{Phases}
  & \multicolumn{1}{|c|}{~\nth{1} Year (2008)}
    & \multicolumn{1}{|c|}{~\nth{2} Year (2009)}
      & \multicolumn{1}{|c|}{~\nth{3} Year (2010)}\\
\hline
\hline
(A) & Design
  & \boxone\boxone\boxone\boxone
    & \boxtwo\nobox\nobox\nobox
      & \nobox\nobox\boxthree\boxthree\\[1ex]
(B) & Implementation
  & \nobox\boxone\boxone\boxone
    & \boxone\boxone\boxone\boxone
      & \boxtwo\nobox\boxthree\boxthree \\[1ex]
(C) & Integration
  & 
    & \nobox\boxtwo\boxone\boxone
      & \boxone\boxtwo\boxthree\boxthree \\[1ex]
(D) & Evaluation
  & \nobox\nobox\nobox\boxone
    & \boxone\boxone\boxone\boxone
      & \boxone\boxone\boxthree\boxthree \\[1ex]
\hline
\hline
\multicolumn{2}{|c}{Legend:} & \multicolumn{3}{l|}{\footnotesize \boxone\nobox Allocated time period} \\
\multicolumn{2}{|c}{}        & \multicolumn{3}{l|}{\footnotesize \boxtwo\nobox Time period to be allocated in special circunstances} \\
\multicolumn{2}{|c}{}        & \multicolumn{3}{l|}{\footnotesize \boxthree\nobox Time period outside the scope of the project} \\
\hline
\end{tabular}
\end{center}

\subsection{Experiments involving humans or humans materials (Untersuchungen am
Menschen oder an vom Menschen entnommenem Material)}

Not applicable.

\enlargethispage{2ex}

\subsection{Experiments with animals (Tierversuche)}

Not applicable.

\subsection{Experiments with recombinant DNA (Gentechnologische Experimente)}

Not applicable.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% BEANTRAGTE MITTEL
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Funds requested (Beantragte Mittel)}

The following are the funds requested for this project for the period of 30
months.

\subsection{Staff (Personalkosten)}

The scientific staff requested for this project includes:

\begin{itemize}
\item 1 post-doctorand researcher (wiss.\ Mitarbeiter E14 TV-L-Äquivalent)
        for 18 months (Jan 2008 -- June 2009)
\item 1 post-doctorand researcher (wiss.\ Mitarbeiter E14 TV-L-Äquivalent)
        for 21 months (Oct 2008 -- June 2010)
\item 2 programmers (stud.\ Hilfskraft, 10 h/Woche) for 27 months each
        (Apr 2008 -- June 2010)
\end{itemize}

{\bf TODO Jens: update}

Zur Durchführung des Projekts wird die Stelle für einen wissenschaftlichen
Mitarbeiter (Postdoktorand) beantragt.
Hierbei handelt es sich im Idealfall um einen promovierten Informatiker
mit einschlägigen Vorarbeiten
im Bereich des Algorithm Engineering und der Sequenzanalyse.
Unterstützt werden soll er durch eine studentische Hilfskraft,
insbesondere bei Routineaufgaben im Bereich der Implementierung und Evaluation.

Für die Postdoktoranden-Stelle steht mit Klaus-Bernd Schürmann
ein hochqualifizierter Kandidat zur Verfügung,
der voraussichtlich im Sommer 2007 seine Dissertation
mit dem Thema ``Suffix Arrays in Theory and Practice''
an der Technischen Fakultät der Universität Bielefeld abschließen wird.
Aus den Bielefelder Studiengängen
\textit{Naturwissenschaftliche Informatik} (Bachelor/Master, Diplom)
und \textit{Bioinformatik und Genomforschung} (Bachelor/Master)
stehen genügend gut qualifizierte Studierende als
studentische Hilfskräfte zur Verfügung.

\bigskip
\begin{tabular*}{\textwidth}{@{\extracolsep{\fill}}llrl}
\textbf{Summary:}          & \nth{1} Year (1. Jahr) & 15 Months (Monate) & wiss.\ MA E14 \\
\textbf{(Zusammenfassung)} &                        & 18 Months (Monate) & stud.\ HK, 10 h/Woche\\
                           & \nth{2} Year (2. Jahr) & 18 Months (Monate) & wiss.\ MA E14 \\
                           &                        & 24 Months (Monate) & stud.\ HK, 10 h/Woche \\
                           & \nth{3} Year (3. Jahr) & 06 Months (Monate) & wiss.\ MA E14 \\
                           &                        & 12 Months (Monate) & stud.\ HK, 10 h/Woche
\end{tabular*}

\subsection{Scientific instrumentation (Wissenschaftliche Geräte)}

Because of the existing computer infrastructure of the Genome Informatics group
as well as the guarantee of maintenance of this equipment by the Bioinformatics
Resource Facility of the CeBiTec, no funds for compuer equipment are requested
for this project.

{\bf TODO Jens: check possibility of memory upgrade}

Durch die vorhandene gute Rechnerausstattung der AG Genominformatik
sowie die Sicherstellung der Wartung dieser Ausstattung
durch die Bioinformatics Resource Facility des Centrums für Biotechnologie
(CeBiTec)
werden keine Mittel für die Ausstattung von Computerarbeitsplätzen beantragt.
Da die vorhandene Rechnerarchitektur allerdings sehr homogen auf Basis
von Produkten der Firma SUN aufgebaut ist,
die entwickelten Methoden aber auch auf alternativen
Plattformen, insbesondere handelsüblichen PCs,
getestet und optimiert werden sollen,
sollen im Laufe des ersten Projektjahres
drei solche Rechner mit aktueller Ausstattung
angeschafft werden, vermutlich je ein Produkt der Hersteller
AMD, Intel, Apple/Macintosh.
Hierfür werden je 2.500,00 Euro, insgesamt also 7.500,00 Euro beantragt.

\bigskip
\setlength{\tabcolsep}{2mm}
\textbf{Summary (Zusammenfassung):}\hfill\begin{tabular}[t]{lr}
\nth{1} Year (1. Jahr) &     0,00 EUR \\
\nth{2} Year (2. Jahr) &     0,00 EUR \\
\nth{3} Year (3. Jahr) &     0,00 EUR \\[.1ex]
\hline
\multicolumn{1}{l}{\textbf{Sum (Summe) 4.2}} & \textbf{0,00 EUR}
\end{tabular}

\subsection{Consumables (Verbrauchsmaterial)}

No funds for consumables are requested. The materials needed in this project,
including office material, will be provided from the budget of the Genome
Informatics research group.

\bigskip
\setlength{\tabcolsep}{2mm}
\textbf{Summary (Zusammenfassung):}\hfill\begin{tabular}[t]{lr}
\nth{1} Year (1. Jahr) & 0,00 EUR \\
\nth{2} Year (2. Jahr) & 0,00 EUR \\
\nth{3} Year (3. Jahr) & 0,00 EUR \\[.1ex]
\hline
\multicolumn{1}{l}{\textbf{Sum (Summe) 4.3}} & \textbf{0,00 EUR}
\end{tabular}

\subsection{Travel (Reisen)}

Although co-operation meetings with Dr. Felipe da Silva and his team are likely
to be necessary during the course of the project, especially during the
evaluation phase (beginning of \nth{2} year), no funds for these travels are
requested, as these will be funded from other sources. Beyond that, one or two
intercontinental or national/European conference attendances of the scientific
researches are planned per year, as detailed in the following table:

\begin{center}
\setlength{\tabcolsep}{2mm}
\begin{tabular}{lllr}
\hline
wiss.\ MA E14           & \nth{1} Year & 1 Intercontinental conference & 2.000,00 EUR \\
(Jan 2008 -- June 2009) &              & 1 European conference         & 1.500,00 EUR \\
                        & \nth{2} Year & 1 Intercontinental conference & 2.000,00 EUR \\
                        & \nth{3} Year &                               &     0,00 EUR \\
\hline
wiss.\ MA E14           & \nth{1} Year &                               &     0,00 EUR \\
(Oct 2008 -- June 2010) & \nth{2} Year & 1 Intercontinental conference & 2.000,00 EUR \\
                        &              & 1 European conference         & 1.500,00 EUR \\
                        & \nth{3} Year & 1 Intercontinental conference & 2.000,00 EUR \\
\hline
\end{tabular}
\end{center}
Example of intercontinental conferences:\\
\hspace*{1em}ISMB: \textit{Conference on Intelligent Systems for Molecular Biology}
  (2008 in Toronto, Canada)\\
\hspace*{1em}RECOMB: \textit{Conference on Research in Computational Molecular Biology}
  (2008 in Singapore)\\
\hspace*{1em}WABI: \textit{Workshop on Algorithms in Bioinformatics}
  (2007 in Philadelphia, USA)\\

Example of national and European conferences:\\
\hspace*{1em}ECCB: \textit{European Conference on Computational Biology}
  (2008 in Cagliary, Italy)\\
\hspace*{1em}GCB: \textit{German Conference on Bioinformatics}
  (2007 in Potsdam)\\

\bigskip
\setlength{\tabcolsep}{2mm}
\textbf{Summary (Zusammenfassung):}\hfill\begin{tabular}[t]{lr}
\nth{1} Year (1. Jahr) & 3.500,00 EUR \\
\nth{2} Year (2. Jahr) & 5.500,00 EUR \\
\nth{3} Year (3. Jahr) & 2.000,00 EUR \\[.1ex]
\hline
\multicolumn{1}{l}{\textbf{Sum (Summe) 4.4}} & \textbf{11.000,00 EUR}
\end{tabular}

\subsection{Publication expenses (Publikationskosten)}

The applicant declares the goal of concentrating the selection of its
publication media on \textit{open access}, which frequently incurs in
publication fees. Nevertheless, no means for publication costs are requested for
this project since the library of the University of Bielefeld supports this
procedure through corporate memberships of \textit{open-access} publishers such
as BioMed Central.

\bigskip
\setlength{\tabcolsep}{2mm}
\textbf{Summary (Zusammenfassung):}\hfill\begin{tabular}[t]{lr}
\nth{1} Year (1. Jahr) & 0,00 EUR \\
\nth{2} Year (2. Jahr) & 0,00 EUR \\
\nth{3} Year (3. Jahr) & 0,00 EUR \\[.1ex]
\hline
\multicolumn{1}{l}{\textbf{Sum (Summe) 4.5}} & \textbf{0,00 EUR}
\end{tabular}

\subsection{Other costs (Sonstige Kosten)}

None.

\bigskip
\setlength{\tabcolsep}{2mm}
\textbf{Summary (Zusammenfassung):}\hfill\begin{tabular}[t]{lr}
\nth{1} Year (1. Jahr) & 0,00 EUR \\
\nth{2} Year (2. Jahr) & 0,00 EUR \\
\nth{3} Year (3. Jahr) & 0,00 EUR \\[.1ex]
\hline
\multicolumn{1}{l}{\textbf{Sum (Summe) 4.6}} & \textbf{0,00 EUR}
\end{tabular}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% VORAUSSETZUNGEN
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Prerequisites for carrying out the project (Voraussetzungen für die
Durchführung des Vorhabens)}

\subsection{Your team (Zusammensetzung der Arbeitsgruppe)}

The research group Genome Informatics was founded in March 2002, with financial
support from the ``Bioinformatics'' DFG Initiative, being today one of the
central groups of the Bielefeld Institute for Bioinformatics. Two junior
research groups are part of the group: \emph{Computation Methods for Emerging
Technologies} (Dr. Sven Rahmann; granted by the Bielefeld University), and
\emph{Combinatorial Search Algorithms in Bioinformatics} (Dr. Ferdinando
Cicalese; granted by the Sofja-Kovalevskaja-Prize from the
Alexander-von-Humboldt Foundation). A former junior research group, \emph{
Computer-Science Methods for Mass Spectrometry} (Prof. Dr. Sebastian B\"{o}cker;
granted by the DFG-Action Plan Infor\-matic/Emmy-Noether-Program), was dissolved
as the group leader was promoted to professor at the University of Jena. The
students of this group joined the other groups to continue their works.

\subsection{Cooperation with other scientists (Zusammenarbeit mit anderen
Wissenschaftlerinnen und Wissenschaftlern)}
\enlargethispage{3ex}

The applicant has several ongoing cooperations in diverse research areas, with
the following ones being in direct relation to this application:

Prof.\ Stefan Kurtz, Universität Hamburg and Prof.\ Robert Giegerich,
Universität Bielefeld: cooperation on engineering of algorithms for
efficient construction of suffix arrays.

Dr.\ Gene Myers, Howard Hughes Medical Institute/Janelia Farm:
cooperation on engineering of algorithms for index-based, non-heuristic,
approximate text search.

Prof.\ Bernd Weisshaar, Chair of Genome Research, Universität Bielefeld:
cooperation on large-scale repeat analysis with the goal of better primer
design for EST detection in the genome of the sugar beet
\textit{Beta vulgaris}.

We also plan to start a cooperation with Dr.~Felipe Rodrigues da~Silva from
Embrapa, the Brazilian National Agricultural Research Institution. Being
involved in several genome projects in Brazil at present and in the past,
Dr.~da~Silva will benefit from new algorithmic methods to be developed in this
project to deal with the large amounts of data that will be produced in the near
future. He will also contribute to the project with his expertise in genome
projects and his extensive experience in genome assembly and annotation.

%% Begin ignore block
\ignore{
Dr.\ Veli Mäkinen, Universität Helsinki:
Gemeinsame Betreuung zweier
Diplomarbeiten mit den Themen
\textit{Kompressionsverstärkung für Textdaten unter Benutzung der
Burrows-Wheeler-Transformation} (Peter Husemann; abgeschlossen im Sommer 2006)
und
\textit{Dynamic FM-Index with application to space-efficient construction of
Compressed Suffix Arrays}
(Wolfgang Gerlach; Fertigstellung voraussichtlich im März 2007).
Diese Kooperation begann im akademischen Jahr 2004/2005,
als Dr.\ Mäkinen als Postdoktorand in der Arbeitsgruppe Genominformatik
in Bielefeld beschäftigt war
und soll im Rahmen dieses Projekts weiter ausgebaut werden.
Eine Bestätigung der Kooperationsbereitschaft von Dr.\ Mäkinen ist dem
Antrag in der Anlage beigefügt.

Prof.\ Stefan Kurtz, Universität Hamburg und
Prof.\ Robert Giegerich, Universität Bielefeld:
Gemeinsame Arbeiten zum Engineering von Algorithmen zur effizienten
Konstruktion von Suffixbäumen
und (mit Prof.\ Kurtz) gemeinsame Betreuung einer Diplomarbeit mit dem Thema
\textit{Flexible Mustersuche unter Einsatz der Burrows-Wheeler-Transformation}
(Andrea Papst; abgeschlossen im Oktober 2005).

Dr.\ Gene Myers, Howard Hughes Medical Institute/Janelia Farm:
Gemeinsame Arbeiten zum Engineering von Algorithmen zur indexbasierten,
nicht-heuristischen approximativen Textsuche.

Über diese durch gemeinsame Publikationen bzw.\ gemeinsam betreute
Abschlussarbeiten
belegten Kooperationen hinaus hat der Antragsteller Gespräche mit
Prof.\ Knut Reinert, FU Berlin, geführt zur Auslotung von
Möglichkeiten der Einbindung von in Bielefeld entwickelten
Implementierungen von Indexstrukturen zur Textsuche in die
in der Arbeitsgruppe von Prof.\ Reinert entwickelte
Softwarebibliothek \textit{SeqAn} für die Sequenzanalyse.
Diese Gespräche sind sehr erfolgreich verlaufen und im Erfolgsfall
des vorliegenden Antrags soll die Zusammenarbeit konkretisiert werden,
da die Expertisen und Vorarbeiten beider Arbeitsgruppen sich in idealer
Weise ergänzen.

Weitere aktive Kooperationen des Antragstellers, die nicht mit dem vorliegenden
Antrag in Beziehung stehen, existieren mit
Prof.\ Anne Bergeron (Université du Québec à Montréal),
Dr.\ Benno Schwikowski (Institut Pasteur, Paris),
Dr.\ Gilles Didier (Les Universités à Marseille),
Prof. Steffen Heber (North Carolina State University),
Dr.\ Michael Sammeth (Institut Municipal d'Investigació Mèdica, Barcelona)
und Dr.\ Dekel Tsur (Ben Gurion University, Israel).
}
%% End ignore block

\subsection{Foreign contacts and collaborations (Arbeiten im Ausland und
Kooperation mit Partnern im Ausland)}

As mentioned in the previous item, we plan to cooperate with Dr.~Felipe
Rodrigues da~Silva from Embrapa, the Brazilian National Agricultural Research
Institution, and this cooperation may involve visits from researchers of
both sides.

\subsection{Scientific equipment available (Apparative Ausstattung)}

In early 2003, the Genome Informatics research group aquired server and
workstations from Sun Microsystems that were integrated in the CeBiTec
infrastructure provided by the Bioinformatics Resource Facility (BRF), where
they are available for use by any CeBiTec member. As a result, the members of
the Genome Informatics group have complete access to the BRF infrastructure,
including a cluster with 472 CPUs, a 10-Gigabit Ethernet network, a
44-Terabyte Backup server, and various computer servers with up to 96
GB of main storage. According to statement of the BRF director, the
existing infrastructure is sufficient for the requirements of the project
proposed here. Personal computers for the scientific staff and students are
available from the Genome Informatics research group resources.

%% Begin ignore block
\ignore{
Im Rahmen der Erstausstattung der AG Genominformatik wurden im Frühjahr 2003
leistungs\-fähige Server und Workstations der Firma SUN angeschafft.
Diese wurden in die von der "`Bioinformatics Resource Facility"' (BRF)
des CeBiTec betreute Infrastruktur integriert, wo sie allen Mitgliedern
des CeBiTec zur Verfügung stehen. Im Gegenzug partizipiert die Arbeitsgruppe
des Antragstellers an der von anderen Arbeitsgruppen zur Verfügung gestellten
Infrastruktur, insbesondere einem verteilten Cluster mit 472 CPUs,
einem leistungsfähigen 10 Gigabit Ethernet-Netzwerk
und einem 44 Terabyte Backup
sowie diversen Compute-Servern mit bis zu 96 GByte Hauptspeicher.
Nach Aussage des Leiters der BRF wird die vorhandene Kapazität
für die von dem hier beantragten Projekt benötigten
Rechenleistungen ausreichen.
Arbeitsplatzrechner für die wissenschaftlichen Mitarbeiter und studentischen
Hilfskräfte stehen ebenfalls
noch aus der Erstausstattung der Arbeitsgruppe zur Verfügung.
}
%% End ignore block

\subsection{Your institution's general contribution (Laufende Mittel für
Sachausgaben)}

The Genome Informatics group has the necessary infrastructure of a scientific
working group with office rooms, office equipment and a secretary. The project
requested here will benefit from this infrastructure made available at a value,
roughly estimated, of 1,000 EUR annually (including office material and travel
expenses).

%% Begin ignore block
\ignore{
Die AG Genominformatik ist 
mit der notwendigen Infrastruktur einer wissenschaftlichen
Arbeitsgruppe (Sekretariat, Verbrauchsmaterialien, Reisekosten der Mitarbeiter)
hinreichend gut ausgestattet.
Dem Projekt werden damit
Sachmittel (in Form von Verbrauchsmaterialien, Reisekosten des
Antragstellers, etc.) in Höhe von (grob geschätzt) 1.000 EUR jährlich
zur Verfügung gestellt.
}
%% End ignore block

\subsection{Conflicts of interest with economic activities (Interessenkonflikte
bei wirtschaftlichen Aktivitäten)}

\enlargethispage{4.5ex}

None.

\subsection{Other requirements (Sonstige Voraussetzungen)}

None.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% ERKLÄRUNGEN
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Declaration (Erklärungen)}

A request for funding this project has not been submitted to any other
addressee. In the event that I submit such request, I will inform the Deutsche
Forschungsgemeinschaft immediately.

The DFG liaison officer of the Bielefeld University, Herr Prof.\ Egelhaaf,
was informed about this application.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% UNTERSCHRIFT
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Signature (Unterschrift)}
\thispagestyle{empty}

\vspace{2ex}Bielefeld, 19. August 2008

\vspace{2ex}\hspace{17em}(Prof.\ Jens Stoye)

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% ANLAGEN
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{List of attachments (Verzeichnis der Anlagen)}

\begin{enumerate}
\item Curriculum vitae of the applicant (DFG-Vordruck 10.04)
\item Complete publication list of the applicant
\item Letter of co-operation agreement from Dr.\ Felipe Rodrigues da Silva,
      Embrapa, Brazil
\item CD-ROM with electronic versions of all request documents
\end{enumerate}

All attachments do not need to be returned.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% APPENDIX
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\appendix

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% LITERATUR
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{References (Literatur zum Antrag)}

\renewcommand{\refname}{\vspace*{-4ex}}
\bibliographystyle{abbrv}
\bibliography{biblio}

\end{document}
