\documentclass[10pt]{article}
\usepackage{a4wide}
\usepackage[german,english]{babel}
\usepackage[latin1]{inputenc}
\usepackage{url}
\usepackage{epsfig}
\usepackage[super]{nth}

%\usepackage{times} % okay (KURZ)
%\usepackage{charter} % okay (MITTEL)
\usepackage{palatino} % okay (LANG)

\parindent0em
\parskip1ex

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% MACROS AND DEFINITIONS
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newcommand{\nobox}{\setlength{\unitlength}{1.1mm}%
                      \begin{picture}(10,2)
                      \end{picture}}
\newcommand{\boxone}{\setlength{\unitlength}{1.1mm}%
                     \begin{picture}(10,2)
                       \put(0,0){\rule{11mm}{2.2mm}}
                     \end{picture}}
\newcommand{\boxtwo}{\setlength{\unitlength}{1.1mm}%
                     \begin{picture}(10,2)
                       \put(0,0.0){\line(1,0){10}}
                       \put(0,0.5){\line(1,0){10}}
                       \put(0,1.0){\line(1,0){10}}
                       \put(0,1.5){\line(1,0){10}}
                       \put(0,2){\line(1,0){10}}
                       \put(0,0){\line(0,1){2}}
                       \put(10,0){\line(0,1){2}}
                     \end{picture}}
\newcommand{\boxthree}{\setlength{\unitlength}{1.1mm}%
                     \begin{picture}(10,2)
                       \put(0,0){\line(1,0){10}}
                       \put(0,0){\line(0,1){2}}
                       \put(0,2){\line(1,0){10}}
                       \put(10,0){\line(0,1){2}}
                       \multiput(0,0)(1.7,0){4}{\line(5,2){4.8}}
                       \multiput(4.8,0)(1.7,0){4}{\line(-5,2){4.8}}
                     \end{picture}}

\newcommand{\ignore}[1]{}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% TITELSEITE
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}

\thispagestyle{empty}
\vspace*{20ex}
\begin{center}
\large
{\bf Antrag an die Deutsche Forschungsgemeinschaft}\\
Kennedyallee 40, 53175 Bonn\\[5ex]
auf Gewährung einer Sachbeihilfe\\[10ex]
{\bf Dual Sparse De-Bruijn Subgraphs (DSDS)\\
for High-throughput Sequencing}\\[5ex]
Prof.\ Dr.\ Jens Stoye\\
Arbeitsgruppe Genominformatik\\[2ex]
Technische Fakultät der Universität Bielefeld\\[2ex]
und\\[2ex]
Institut für Bioinformatik\\
Centrum für Biotechnologie der Universität Bielefeld
\end{center}
\newpage
\thispagestyle{empty}~
\newpage
\setcounter{page}{1}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% ALLGEMEINE ANGABEN
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{General information (Allgemeine Angaben)}

Proposal for a research grant (new application).

\subsection{Applicant (Antragsteller)}

\setlength{\tabcolsep}{0em}

\begin{tabular}{l@{~~~~~~}l}
Name:              & Prof.\ Dr.\ Jens Stoye\\
Employment status: & Universitätsprofessor (C4)\\
Date of birth:     & 17.\ 03.\ 1970\\
Nationality:       & german\\
Institution:       & Technische Fakultät und Institut für Bioinformatik\\
                   & der Universität Bielefeld\\
Office address:    & Universität Bielefeld\\
                   & Technische Fakultät\\
                   & AG Genominformatik\\
                   & 33594 Bielefeld\\
Telephon:          & 0521 - 106 - 3852\\
Telefax:           & 0521 - 106 - 6495\\
E-Mail:            & stoye@techfak.uni-bielefeld.de\\
Private address:   & Kesselstraße 3\\
                   & 33602 Bielefeld\\
                   & 0521 -- 9675479
\end{tabular}

See attached documents for a tabular CV.

\subsection{Topic (Thema)}

Watson-Crick complement invariant version of sparse De-Bruijn subgraphs for
analysis of high-throughput sequencing data.

%% 120 characters
%% Max allowed: 140 characters

Watson-Crick-Komplement-unabhängige Version von sparse De-Bruijn-Teilgraphen
für die Analyse von Hochdurchsatz-Sequenzierdaten.

%% 127 characters
%% Max allowed: 140 characters

\subsection{Scientific discipline and field of work (Fachgebiet und
Arbeitsrichtung)}

Scientific discipline: Computer Science, algorithm engineering\\
Field of work: Sequence analysis

\subsection{Scheduled total duration (Voraussichtliche Gesamtdauer)}

Duration: 30 months

\subsection{Application period (Antragszeitraum)}

First proposal: Yes\\
Time period: \nth{1} January 2008 -- \nth{30} June 2010\\
Funding commence: \nth{1} January 2008

\subsection{Summary (Zusammenfassung)}

{\bf TODO: join two paragraphs? (max. 15 lines)}

High-throughput DNA sequencing methods in use today are able to produce
staggering amounts of data on a daily basis that demand extensive computing
resources to assemble, finish and process genomic information. The Center for
Biotechnology (CeBiTec) at Bielefeld University will soon have its own
high-throughput sequencing machine, a patented technology developed by 454 Life
Sciences, which is capable of producing millions of raw DNA bases per hour.
However, what makes sequencing a challenge today is the subsequent computational
problem of correctly assembling the reads into the original sequence.

This project aims to develop a novel approach to genome assembly based on Dual
Sparse De-Bruijn Subgraphs (DSDS). De-Brujin subgraphs are already being studied
by members of our research group. The new approach consists of a version of our
current model that exploits the Watson-Crick complementarity nature of the DNA.
The project consists of two parts. First, we will study and develop the new
graph, and design the necessary data structures for efficient implementation. In
the second part of the project, we will apply our new approach in practice to
genomic assembly of DNA data sequenced at the CeBiTec and other institutions
and, extending ongoing research, to detect repeats in these data.

%% 15 lines, 1285 characters
%% Max allowed: 15 lines, 1600 characters

\subsection*{Zusammenfassung}

{\bf TODO: reduce to 1600 characters (currently 1650) and 15 lines
(currently 19)}

Hochdurchsatztechniken, wie sie mittlerweise immer mehr Verbreitung in
der DNA-Sequenzierung finden, erlauben die Produktion atemberaubend
großer genomischer Sequenzdatenmengen in kurzer Zeit, was hohe
Anforderungen auch für die bioinformatische Auswertung mit sich bringt,
sei es bei der Assemblierung, dem \textit{Finishing} oder der
Weiterverarbeitung der genomischen Information. Das Zentrum für
Biotechnologie (CeBiTec) der Universität Bielefeld wird in Kürze eine
eigene Hochdurchsatz-Sequenziermaschine der Firma Roche/454 in Betrieb
nehmen, mit einer theoretischen Kapazität von stündlich mehreren
Millionen DNA-Basenpaaren. Die große Herausforderung beim Einsatz
dieser Technologie ist allerdings die nachfolgende bioinformatische
Analyse, insbesondere die Assemblierung der gelesenen Teilsequenzen in
das ursprüngliche Genom.

Ziel des hier beantragten Projekts ist die Entwicklung einer neuen
Methode zur Genomassemblierung auf der Basis dualer
speicherplatzsparender De-Bruijn-Subgraphen (DSDS). De-Bruijn-Subgraphen
werden bereits in unserer Arbeitsgruppe untersucht. Neuartig an dem hier
vorgeschlagenen Ansatz ist eine Erweiterung des Modells, das die
Watson-Crick-Komplementarität von DNA-Sequenzen berücksichtigt. Das
Projekt besteht aus zwei Teilen. Im ersten Teil soll die neue
Graph-Datenstruktur definiert und theoretisch untersucht sowie effizient
implementiert werden. Im zweiten Teil soll der neue Ansatz in der Praxis
erprobt werden, zum einen bei der Assemblierung genomischer DNA, die am
CeBiTec und an anderen Institutionen sequenziert wird, und zum anderen
zur \textit{Repeat}-Analyse dieser Daten, als Erweiterung bereits
bestehender Forschungen.

%% 14 lines, 1650 characters
%% Max allowed: 15 lines, 1600 characters

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% STAND DER FORSCHUNG, EIGENE VORARBEITEN
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{State of the art, preliminary work (Stand der Forschung, eigene
Vorarbeiten)}

\subsection{State of the art (Stand der Forschung)}

De-Bruijn graphs were first studied in the end of the \nth{19} century,
although in an implicit form, and were formally defined in 1946 by N.~G.~de~
Bruijn~\cite[Chapter 3]{moreno}. They are directed graphs with a simple and
clear definition that are easy to build, and that have interesting properties:
they are regular, have small diameter, and are both Hamiltonian and Eulerian. As
a result, De-Bruijn graphs have been used in diverse applications such as
network models, pseudo-random number generation, and DNA analysis~\cite{moreno}.
Similar graphs have also been used for the identification of repeat
families~\cite{raphael04}, but the first use in Bioinformatics was probably the
Eulerian path approach to sequence assembly proposed by Idury and
Waterman~\cite{idury95} and extended by Pevzner, Tang and
Waterman~\cite{pevzner01a}.

Despite the success achieved by the sequence assembler developed by Pevzner and
co-workers for bacterial genomes, De-Bruijn graphs have not been further
explored in computational biology. In some works they appear slightly modified,
and are more often used as a multiple alignment displayer than a basic
tool~\cite{pevzner04,raphael04}. As Myers~\cite{myers05} points out, the main
problem with De-Bruijn graphs is that, in the original definition, they are
space inefficient.

A $d$-dimensional De-Bruijn Graph $G=(V,E)$ on an alphabet $\Sigma$ is a
directed graph defined as follows:
\begin{eqnarray*}
   V &=& \Sigma^d \\ 
   E &=& \lbrace (u,v)\ |\ u,v \in V \mbox{ and }
   u_{i+1} = v_i, \mbox{ for all } 1 \leq i < d \rbrace,
\end{eqnarray*}
%%
where $u_i$ is the $i$-th character of the string represented by $u$. Strings of
length at least $d$ over the same alphabet describe walks on the $d$-dimensional
De-Bruijn graph. Given a set of strings, we define the associated
$d$-dimensional \emph{De-Bruijn subgraph} as the subgraph of the $d$-dimensional
De-Bruijn graph that contains all the walks described by these strings but no
extra vertex or arc.

Sequence-associated De-Bruijn subgraphs have a nice asymptotic behavior: The
maximum number of nodes increases linearly with the size of the input and
decreases with the dimension $d$ of the graph. The main problem is that,
although these graphs scale well with the string set size, graphs corresponding
to even relatively small genomes (such as of bacteria) are already prohibitively
large for an average computer. This could probably explain why De-Bruijn graphs
have not been further used in Bioinformatics.

Another disadvantage of simple De-Bruijn graphs for genome assembly is that they
require cutting the sequence reads into small pieces to finally build the graph.
Myers~\cite{myers05} suggested that this step might not be necessary. In
fact, we have successfully constructed a hybrid structure based on De-Bruijn
subgraphs without explicitly using sequence comparisons, with the added
advantage of allowing the representation of a whole genome of a higher-order
organism in the memory of a typical personal computer.

In a typical sequence-associated De-Bruijn subgraph, the graph branches are more
important than the nodes in non-branching paths, and most of the nodes in such a
graph are exactly in non-branching paths. Hence, the first step to reduce its
size is to contract non-branching paths into single nodes containing sequences
of unlimited length, similarly to what Idury and Waterman~\cite{idury95} have
done in their assembly algorithm.

In order to avoid constructing the large non-contracted graph, we define an
indexed structure, called \emph{sequence graph}, formed by a relaxed definition
of a De-Bruijn graph and a mapping from $d$-tuples to the nodes that represent
them. In the relaxed De-Bruijn graph, the length of a sequence in a node is
unbounded. This allows us to create a compact representation of sparse De-Bruijn
subgraphs without non-branching paths, while the access to a particular
$d$-tuple is still possible via the index. Our initial experiments show that the
sequence graph is able to store the same information as a De-Bruijn subgraph
using roughly $25\%$ of the memory. We also notice a proportional reduction of
the time spent in constructing the graph.

We believe that the sequence graph can be further improved by exploiting a
peculiarity of genomic data, namely, the reverse-complementarity nature of DNA.
A DNA molecule is formed by two complementary strands that bind according to
pairing rules in the double-helix structure discovered by Watson and Crick,
namely guanine (G) pairs with cytosine (C), and adenine (A) pairs with thymine
(T). Current sequencing methods are not able to distinguish from which strand a
given sequence comes and, as a result, each sequence read is represented twice
in the sequence graph.
Our aim is to avoid this duplication by adding the concept of
complementarity both in the index and in the relaxed De-Bruijn graph. The
structure we propose is called Dual Sparse De-Bruijn Subgraph. In case such a
structure can be efficiently built and updated, we will be able to double the
efficiency of the structure we already have both in terms of memory requirements
and construction time.

\subsection{Preliminary work, progress report (Eigene Vorarbeiten,
Arbeitsbericht)}

The proposed data structure has several applications in genome
analysis, which is the main subject of most of the activities
in our Genome Informatics research group,
as can be seen below.

\paragraph{Sequence Searching.}
The group worked in close collaboration with the research group for
Practical Computer Science in Bielefeld, headed by Prof.\ Robert Giegerich,
on indexing methods for large amounts of
genomic data. Among them are the \emph{suffix
  trees}~\cite{GIE-KUR-STO-2003}, which are at the moment very popular in
the field of Bioinformatics.  Suffix trees can be used in many
different applications~\cite{GUS-1997}. The identification of repeated
sequences~\cite{GUS-STO-2004,KUR-CHO-OHL-SCH-STO-GIE-2001,STO-GUS-2002}
is one of them.

\emph{Affix trees}~\cite{STO-2000} are a generalization of suffix
trees for applications related to the search for a given pattern in a
text. They represent a connection between suffix trees and their
coresponding complementary reverse \emph{prefix trees}.  Each
substring of a text is therefore represented twice (as if we read it
from both left to right and right to left). This gives support to
flexible search strategies. The linear memory requirement of affix
trees is known since its development in Bielefeld in 1995.  An
efficient algorithm for constructing them was presented later by
Maa\ss~\cite{MAS-2003}. At the moment we are exploring an application
to the search of special non-homogeneous patterns, where the use of
affix trees can improve considerably the search efficiency.

While suffix trees, and specially affix trees, theoretically allow the
implementation of both time and space efficient algorithms, the amount
of memory required by such structures in practice are not negligible. An
efficient alternative is the use of \emph{suffix arrays}.  They can be
used in almost all scenarios where suffix trees can be applied, and
their representation requires much less memory in
practice~\cite{ABO-KUR-OHL-2004}. An algorithm for the construction of
suffix arrays from genomic data was developed in the Genome Informatics
research group~\cite{SCH-STO-2007}. The algorithm may be used in
many different applications, and has a running-time comparable to the
best existing algorithms, being even faster in many cases.

An even less memory consuming index for database searches is the so
called {\em $q$-gram hash}, where the positions of all $q$-grams
(substrings of length $q$) are stored in a table. Based on this index,
we were able to design a filter algorithm in collaboration with Gene
Myers (Janelia Farms Research Campus of the Howard Hughes Medical
Institute). Besides the filter algorithm, the database search software
tool SWIFT~\cite{RAS-STO-MYE-2006} was also a result of this
collaboration. SWIFT not only requires much less memory than similar
software tools, but is also 25 times faster than the most popular of
them, BLAST, without any loss of sensitivity.

\paragraph{Genome evolution modeling.} Information about the evolution 
of genomes may be obtained from the global structure of the gene
arragements in them. In this case, the corresponding maximum parsimony
assumption, namely the minimum number of rearrangements (inversions,
translocations, transpositions, fusions, and fissions of genome
pieces), is used to transform a genome into another one. The related
mathematical theory was studied by Hannenhalli and
Pevzner~\cite{HAN-PEV-1995B,HAN-PEV-1995C,HAN-1996} during the 1990's.
The highly technical content of their work was at that time a big
barrier to the area. Together with Anne Bergeron (Montreal), we could
make important steps to the simplification of this
theory~\cite{BER-HEB-STO-2002,BER-MIX-STO-2004,BER-MIX-STO-2005,BER-STO-2003B}.
Errors in the original works could also be identified and corrected
during the simplification process~\cite{BER-MIX-STO-2005B}. Another
object studied by our group are gene clusters, which are sets of genes
that appear close to each other in different, not so closely related,
genomes.  In~\cite{HEB-STO-2001B,HEB-STO-2001,SCH-STO-2004} we developed
many different algorithms able to efficiently identify gene clusters
in several genomes. At the moment, this work is being extended and
applied to real data. The results are also being combined to data from
other sources, like gene expression studies.

\paragraph{Systems biology and microarray design.} 
Systems biology explores biological systems and their interactions both
mathematically and algorithmically. One of our aims is the development of
methods for the improvement of data quality, which ideally could be applied just
after or even during the data collection, as well as methods for the integration
of data from different experiments. In collaboration with the chair of Genetics
in Bielefeld, we analyze the regulatory network of {\it Corynebacterium
glutamicum}. Another topic of our research is the design of microarrays, which
consists of optimizing the selection of probes
sequences~\cite{RAH-2003,RAH-GRA-2004}, reducing the length of the synthesis
schedule~\cite{RAH-2003B}, and designing the layout of the array (the
arrangement of the probes on the chip) in order to improve the quality of the
manufacturing process~\cite{CAR-RAH-2006,CAR-RAH-2007}.

\paragraph{Repeat analysis.}

Identifying repetitive elements in the genome of eukaryotes is an
important task both if one is interested in study them and if one
needs to avoid them. Although the sequence of some repetitive elements
may be characterized, which allows repeat identification by
traditional sequence comparison methods, the experience shows that
most of the eukaryotes have very specific families of repetitive
elements. For completely sequenced genomes, the {\em de novo}
identification of such sequences may be done using suffix trees. A
software tool called REPuter~\cite{KUR-CHO-OHL-SCH-STO-GIE-2001} was
developed based on this approach. In a collaboration with Prof.\ Bernd
Weisshaar (Chair of Genome Research, Bielefeld University), we are
developing a De-Bruijn subgraph based method for doing the same with
collections of reads of an incompletely sequenced genome. Also here
the size of De-Bruijn subgraphs is a considerable barrier to the
development of software tools for personal computers.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% ZIELE UND ARBEITSPROGRAMM
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Goals and work schedule (Ziele und Arbeitsprogramm)}

\subsection{Goals (Ziele)}

\begin{figure}
  \epsfxsize=0.6\textwidth
  \begin{center}\epsfbox{graphics}
  \end{center}
  \caption{Memory usage (in gigabytes) for De-Bruijn subgraphs and their
    correponding sequence graphs. The number of sequences corresponds to the
    number of 700 bases length randomly taken subsequences of an
    {\em Arabidopsis thaliana} chromosome.}
  \label{fig.graph}
\end{figure}

\begin{figure}
  \epsfxsize=\textwidth
  \begin{center}\epsfbox{advantages}
  \end{center}
  \caption{Four graphs for assembling the sequences {\tt AAACCC} and
    {\tt TTGTGGG}.}
  \label{fig.advantages}
\end{figure}

While working with a De-Bruijn based approach to repeat finding, we
were able to develop an algorithm for building the sequence graph, the
compact version of a De-Bruijn subgraph, directly from the given set
of sequences. The graph on Figure~\ref{fig.graph} presents the amount
of memory used by De-Bruijn graphs and by the corresponding sequence
graphs. Figure~\ref{fig.advantages} exemplifies how memory can be
saved: The De-Bruijn subgraph containing both the sequences {\tt
  AAACCC} and {\tt TTGTGGG}, with the respective complements
(Fig.~\ref{fig.advantages}.2), contains non-branching paths. Since the
information we are interested in are exactly in the graph branches,
the same information represented by the 16 nodes, 48 characters, and
14 edges (Fig.~\ref{fig.advantages}.2) can be represented by 2
nodes, 20 characters and no edge at all (Fig.~\ref{fig.advantages}.3).

Most bioinformatics applications need to deal with both the
input set and its reverse-complement. Without considering the
reverse-complement, important information can be lost, as
Figure~\ref{fig.advantages}.1 shows. In this case, because only the
input set is considered, one may conclude that the given sequences
belong to independent DNA molecules, whereas
Figure~\ref{fig.advantages}.2 shows that both sequence are indeed part of
different strands of the same DNA molecule. The direct consequence of
this is that in a De-Bruijn graph for the DNA alphabet, the
information usually needs to be represented twice.

The goal of this project is to take advantage of the DNA sequence
complementarity to reduce even more the graph size. In the example
shown in Figure~\ref{fig.advantages}.4, the reduction achieved is of
50\%. Of course this is a very special case. In the real life, we will
be facing many problems not shown in the example, like
self-complementary sequences.

More specifically, the goals of the project are:
\begin{description}
  
\item[(A)] Investigation of the proposed Dual Sparse De-Bruijn Subgraph
  (DSDS).
  
\item[(B)] Implementation of a genome assembler, so that we can compare the
  performance of an application using the new graph to the performance
  of similar applications.
  
  We choose the genome assembly problem for mainly three reasons:
  
  \begin{itemize}
  \item The De-Bruijn graph approach do genome assembly is well
    studied and documented. Not only the assembly, but also a
    sequencing error correction method are described in the
    literature.
  \item From the known applications for De-Bruijn graphs, the genome
    assembly is the only one with which the group has no previous
    experience. Since the application is well documented, it is a good
    chance to expand our group's knowledge without risking delaying
    the project.
  \item Although genome assembling may look like an old fashion
    subject, the appearance of completely new sequencing methods in the
    last years suggests that, in the near future, sequencing projects
    may involve lots of different kinds of sequences, with the most
    variable lengths. In this scenario, the De-Bruijn graph approach
    flexibility in respect of the input sequence lengths will be an
    advantage.
  \end{itemize}
  
\item[(C)] Adaptation of the repeat detection method being developed in our
  group to use DSDS.
  
\item[(D)] Validation of the applications developed. For
  attaining this natural final goal, we count with the collaborations both in
  Bielefeld and in Brazil. At the end, the project must produce, and
  publish, an implementation of DSDSs and two validated applications
  using them.

\end{description}

\subsection{Work schedule (Arbeitsprogramm)}

The project's work schedule consists of four phases, oriented towards the
four goals described in the previous section.
For a graphical overview, see the end of this section.

\subsubsection*{(A) Design}

The first phase of the project consists of studying the properties of the Dual
Sparse De-Bruijn Subgraph (DSDS) and designing the data structures on which
will be built
the genome assembler (phase B) and the integration with our existing
repeat detection functions (phase C). In this phase, the storage requirements
for the data structure will be studied in detail, so that the memory
requirements for the genome assembler and repeat detection functions can be
predicted based on the size of the input (sequencing data).

Among the functions the DSDS data structure shall offer, we point out:

\begin{description}
\item[\sc insertSequence] - Makes the necessary transformations in the
  graph, so that the walk corresponding to a given sequence may be
  found in it.
\item[\sc deleteSequence] - Removes the walk corresponding to a given
  sequence from the graph.
\item[\sc traceSequence] - Returns the nodes corresponding to a
  sequence walk in the order they appear in the sequence.
\item[\sc getNode] - Returns the node representing a given $q$-gram.
\end{description}

In this phase a postdoctoral researcher will lead the design of the necessary
data structures, with a team of two students starting the work on the basic DSDS
implementation three months after the beginning of the design phase. Both the
design and the DSDS implementation phases will continue in parallel as the
implementation is likely to require changes in the design of the data structure.
The design of basic data structures is expected to reach a stable state by
October 2008 and a final version of the DSDS should be ready by the end of the
year, although the first quarter of the second year may be also used in case any
serious design problems arise during the implementation phase.

\subsubsection*{(B) Implementation}

This phase consists of the implementation of the DSDS data structure as desribed
above, and the implementation of a fully-functional genome assembler as a
stand-alone software package.

The DSDS implementation will be carried out by a team of two student programmers
lead by a postdoctoral researcher. An initial working implementation providing
the basic functions should be ready by mid-2008 so that the DSDS design can be
exhaustively evaluated and validated until the end of the first year using real
sequencing data. A final implementation of the DSDS is expected to be ready by
the end of 2008, or by the end of the first quarter of 2009 at the latest.

The implementation of the genome assembler using the DSDS data structure starts
in 2009 with a single programmer working under the supervision of both
postdoctoral researchers (with the other programmer working independently on
phase C). The final version of the genome assembler should be ready by the end
of 2009.

\subsubsection*{(C) Integration of existing repeat detection algorithm with DSDS}

Once the implementation of the basic DSDS data structure becomes stable, it will
be integrated with the existing repeat detection algorithms already developed at
the Genome Informatics research group. This phase will be carried out by a
single programmer under the supervision of both postdoctoral researchers, and
it may be concluded in about six months, or at most nine months. This phase
might start as early as January 2009. However, depending on the amount of work
involved in the development of the genome assembler, it might be necessary to
delay the integration phase by up to three months so that, in the beginning,
both programmers can work on the genome assembler. As soon as the integration
phase in concluded, both students will be free to concentrate on the
implementation of the genome assembler, and also help with the evaluation (phase
D).

\subsubsection*{(D) Evaluation}

The DSDS implementation will be evaluated using the genome assembler built in
phase B and the repeat detection algorithms integrated in phase C\@. This
evaluation will use real sequencing data generated by the Center for
Biotechnology (CeBiTec) in Bielefeld and our partners in Brazil. The validation
of the results will be under the responsibility of both postdoctoral researchers, with
the help of both students and biologists from the Genome Research group of
Prof.\ Bernd Weisshaar and from the group of Dr.\ Felipe Rodrigues da Silva at
Embrapa, in Brazil. Running times and memory requirements will also be measured
in order to compare the results with those obtained with similar software
packages.

The evaluation phase will start as soon as the initial implementation of the
basic DSDS data structure is ready (mid-2008), and it will last until the end of
the project, in parallel with the implementation and integration phases. The
evaluation phase will also consist of publishing the results in journals and
presenting them at international conferences.

\subsubsection*{Summary of work schedule}

\begin{center}
\setlength{\tabcolsep}{0cm}
\begin{tabular}{|@{~~}c@{~~}l@{~~}|p{4.4cm}|p{4.4cm}|p{2.2cm}|}
\hline
\multicolumn{2}{|c|}{Phases}
  & \multicolumn{1}{|c|}{2008}
    & \multicolumn{1}{|c|}{2009}
      & \multicolumn{1}{|c|}{2010 (\nth{1} half)}\\
\hline
\hline
(A) & Design
  & \boxone\boxone\boxone\boxone
    & \boxtwo
      & \\[1ex]
(B) & Implementation & & & \\[1ex]
    & -- DSDS
      & \nobox\boxone\boxone\boxone
        & \boxtwo
          & \\[1ex]
    & -- Assembler
      & \nobox\nobox\nobox\nobox
        & \boxone\boxone\boxone\boxone
          & \boxtwo \\[1ex]
(C) & Integration
  & 
    & \boxtwo\boxone\boxone\boxtwo
      & \\[1ex]
(D) & Evaluation
  & \nobox\nobox\boxone\boxone
    & \boxone\boxone\boxone\boxone
      & \boxone\boxone \\[1ex]
\hline
\hline
\multicolumn{2}{|c}{Legend:} & \multicolumn{3}{l|}{\footnotesize \boxone\nobox Allocated time period} \\
\multicolumn{2}{|c}{}        & \multicolumn{3}{l|}{\footnotesize \boxtwo\nobox Time period to be allocated in special circumstances} \\
\hline
\end{tabular}
\end{center}

\subsection{Experiments involving humans or humans materials (Untersuchungen am
Menschen oder an vom Menschen entnommenem Material)}

Not applicable.

\enlargethispage{2ex}

\subsection{Experiments with animals (Tierversuche)}

Not applicable.

\subsection{Experiments with recombinant DNA (Gentechnologische Experimente)}

Not applicable.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% BEANTRAGTE MITTEL
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Funds requested (Beantragte Mittel)}

The following are the funds requested for this project for the period of 30
months.

\subsection{Staff (Personalkosten)}

The scientific staff requested for this project includes:

\begin{itemize}
\item 1 postdoctoral researcher (wiss.\ Mitarbeiter E14)
        for 18 months (Jan 2008 -- June 2009)
\item 1 postdoctoral researcher (wiss.\ Mitarbeiter E14)
        for 21 months (Oct 2008 -- June 2010)
\item 2 programmers (stud.\ Hilfskraft, 10 h/Woche) for 27 months each
        (Apr 2008 -- June 2010)
\end{itemize}

The main research work of the project will be performed by two
postdoctoral researchers, supported by two student programmers.
The first postdoc will perform the initial design of the DSDS
data structure (phase A), its implementation (phase B.1)
and evaluation (phase D).
The second postdoc will concentrate on the implementation
of the assembler (phase B.2) and the integration of the repeat
detection algorithm (phase C) as well as their evaluation (phase D).
The student programmers will support the postdoctoral researchers,
mainly in the implementation and evaluation phases.

A number of Ph.D.\ students will graduate in the next years
from the International Graduate School in Bioinformatics
and Genome Research and the GK Bioinformatik in Bielefeld,
so we do not expect a shortage of highly qualified applicants
for the postdoctoral positions.
From the Bachelor and Master programs in 
\textit{Naturwissenschaftliche Informatik}
and \textit{Bioinformatik und Genomforschung}
at the Faculty of Technology of Bielefeld University
there will be more than enough well qualified candidates
for the student programmer positions.

\bigskip
\begin{tabular*}{\textwidth}{@{\extracolsep{\fill}}llrl}
\textbf{Summary:}          & \nth{1} Year (1. Jahr) & 15 Months (Monate) & wiss.\ MA E14 \\
\textbf{(Zusammenfassung)} &                        & 18 Months (Monate) & stud.\ HK, 10 h/Woche\\
                           & \nth{2} Year (2. Jahr) & 18 Months (Monate) & wiss.\ MA E14 \\
                           &                        & 24 Months (Monate) & stud.\ HK, 10 h/Woche \\
                           & \nth{3} Year (3. Jahr) & 06 Months (Monate) & wiss.\ MA E14 \\
                           &                        & 12 Months (Monate) & stud.\ HK, 10 h/Woche
\end{tabular*}

\subsection{Scientific instrumentation (Wissenschaftliche Geräte)}

Because of the existing computer infrastructure of the Genome Informatics group
as well as the guarantee of maintenance of this equipment by the Bioinformatics
Resource Facility of the CeBiTec, no funds for computer equipment are requested
for this project.

{\bf TODO Jens: check possibility of memory upgrade}

\bigskip
\setlength{\tabcolsep}{2mm}
\textbf{Summary (Zusammenfassung):}\hfill\begin{tabular}[t]{lr}
\nth{1} Year (1. Jahr) &     0,00 EUR \\
\nth{2} Year (2. Jahr) &     0,00 EUR \\
\nth{3} Year (3. Jahr) &     0,00 EUR \\[.1ex]
\hline
\multicolumn{1}{l}{\textbf{Sum (Summe) 4.2}} & \textbf{0,00 EUR}
\end{tabular}

\subsection{Consumables (Verbrauchsmaterial)}

No funds for consumables are requested. The materials needed in this project,
including office material, will be provided from the budget of the Genome
Informatics research group.

\bigskip
\setlength{\tabcolsep}{2mm}
\textbf{Summary (Zusammenfassung):}\hfill\begin{tabular}[t]{lr}
\nth{1} Year (1. Jahr) & 0,00 EUR \\
\nth{2} Year (2. Jahr) & 0,00 EUR \\
\nth{3} Year (3. Jahr) & 0,00 EUR \\[.1ex]
\hline
\multicolumn{1}{l}{\textbf{Sum (Summe) 4.3}} & \textbf{0,00 EUR}
\end{tabular}

\subsection{Travel (Reisen)}

Although co-operation meetings with Dr. Felipe da Silva and his team are likely
to be necessary during the course of the project, especially during the
evaluation phase (beginning of \nth{2} year), no funds for these travels are
requested, as these will be funded from other sources. Beyond that, one or two
intercontinental or national/European conference attendances of the scientific
researches are planned per year, as detailed in the following table:

\begin{center}
\setlength{\tabcolsep}{2mm}
\begin{tabular}{lllr}
\hline
wiss.\ MA E14           & \nth{1} Year & 1 Intercontinental conference & 2.000,00 EUR \\
(Jan 2008 -- June 2009) &              & 1 European conference         & 1.500,00 EUR \\
                        & \nth{2} Year & 1 Intercontinental conference & 2.000,00 EUR \\
                        & \nth{3} Year &                               &     0,00 EUR \\
\hline
wiss.\ MA E14           & \nth{1} Year &                               &     0,00 EUR \\
(Oct 2008 -- June 2010) & \nth{2} Year & 1 Intercontinental conference & 2.000,00 EUR \\
                        &              & 1 European conference         & 1.500,00 EUR \\
                        & \nth{3} Year & 1 Intercontinental conference & 2.000,00 EUR \\
\hline
\end{tabular}
\end{center}
Example of intercontinental conferences:\\
\hspace*{1em}ISMB: \textit{Conference on Intelligent Systems for Molecular Biology}
  (2008 in Toronto, Canada)\\
\hspace*{1em}RECOMB: \textit{Conference on Research in Computational Molecular Biology}
  (2008 in Singapore)\\
\hspace*{1em}WABI: \textit{Workshop on Algorithms in Bioinformatics}
  (2007 in Philadelphia, USA)\\

Example of national and European conferences:\\
\hspace*{1em}ECCB: \textit{European Conference on Computational Biology}
  (2008 in Cagliary, Italy)\\
\hspace*{1em}GCB: \textit{German Conference on Bioinformatics}
  (2007 in Potsdam)\\

\bigskip
\setlength{\tabcolsep}{2mm}
\textbf{Summary (Zusammenfassung):}\hfill\begin{tabular}[t]{lr}
\nth{1} Year (1. Jahr) & 3.500,00 EUR \\
\nth{2} Year (2. Jahr) & 5.500,00 EUR \\
\nth{3} Year (3. Jahr) & 2.000,00 EUR \\[.1ex]
\hline
\multicolumn{1}{l}{\textbf{Sum (Summe) 4.4}} & \textbf{11.000,00 EUR}
\end{tabular}

\subsection{Publication expenses (Publikationskosten)}

The applicant declares the goal of concentrating the selection of its
publication media on \textit{open access}, which frequently incurs in
publication fees. Nevertheless, no means for publication costs are requested for
this project since the library of the University of Bielefeld supports this
procedure through corporate memberships of \textit{open-access} publishers such
as BioMed Central.

\bigskip
\setlength{\tabcolsep}{2mm}
\textbf{Summary (Zusammenfassung):}\hfill\begin{tabular}[t]{lr}
\nth{1} Year (1. Jahr) & 0,00 EUR \\
\nth{2} Year (2. Jahr) & 0,00 EUR \\
\nth{3} Year (3. Jahr) & 0,00 EUR \\[.1ex]
\hline
\multicolumn{1}{l}{\textbf{Sum (Summe) 4.5}} & \textbf{0,00 EUR}
\end{tabular}

\subsection{Other costs (Sonstige Kosten)}

None.

\bigskip
\setlength{\tabcolsep}{2mm}
\textbf{Summary (Zusammenfassung):}\hfill\begin{tabular}[t]{lr}
\nth{1} Year (1. Jahr) & 0,00 EUR \\
\nth{2} Year (2. Jahr) & 0,00 EUR \\
\nth{3} Year (3. Jahr) & 0,00 EUR \\[.1ex]
\hline
\multicolumn{1}{l}{\textbf{Sum (Summe) 4.6}} & \textbf{0,00 EUR}
\end{tabular}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% VORAUSSETZUNGEN
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Prerequisites for carrying out the project (Voraussetzungen für die
Durchführung des Vorhabens)}

\subsection{Your team (Zusammensetzung der Arbeitsgruppe)}

The research group Genome Informatics was founded in March 2002, with financial
support from the DFG Initiative ``Bioinformatics''. As of now, it forms one of
the central groups of the Bielefeld Institute for Bioinformatics. Two junior
research groups are part of the group: \emph{Computational Methods for Emerging
Technologies} (Dr. Sven Rahmann; granted by Bielefeld University), and
\emph{Combinatorial Search Algorithms in Bioinformatics} (Dr. Ferdinando
Cicalese; granted by the Sofja-Kovalevskaja-Prize from the
Alexander-von-Humboldt Foundation). A former junior research group, \emph{
Computer-Science Methods for Mass Spectrometry} (Prof. Dr. Sebastian B\"{o}cker;
granted by the DFG-Action Plan Infor\-matics/Emmy-Noether-Program), was dissolved
as the group leader was promoted to professor at the University of Jena. The
students of this group joined the other groups to continue their works.

\subsection{Cooperation with other scientists (Zusammenarbeit mit anderen
Wissenschaftlerinnen und Wissenschaftlern)}
\enlargethispage{3ex}

The applicant has several ongoing cooperations in diverse research areas, with
the following ones being in direct relation to this application:

Prof.\ Stefan Kurtz, University of Hamburg and Prof.\ Robert Giegerich,
Bielefeld University: cooperation on engineering of algorithms for
efficient construction of suffix arrays.

Dr.\ Gene Myers, Howard Hughes Medical Institute/Janelia Farm:
cooperation on engineering of algorithms for index-based, non-heuristic,
approximate text search.

Prof.\ Bernd Weisshaar, Chair of Genome Research, Bielefeld University:
cooperation on large-scale repeat analysis with the goal of better primer
design for EST detection in the genome of the sugar beet
\textit{Beta vulgaris}.

We also plan to start a cooperation with Dr.~Felipe Rodrigues da~Silva from
Embrapa, the Brazilian National Agricultural Research Institution. Being
involved in several genome projects in Brazil at present and in the past,
Dr.~da~Silva will benefit from new algorithmic methods to be developed in this
project to deal with the large amounts of data that will be produced in the near
future. He will also contribute to the project with his expertise in genome
projects and his extensive experience in genome assembly and annotation.

\subsection{Foreign contacts and collaborations (Arbeiten im Ausland und
Kooperation mit Partnern im Ausland)}

As mentioned in the previous item, we plan to cooperate with Dr.~Felipe
Rodrigues da~Silva from Embrapa, the Brazilian National Agricultural Research
Institution, and this cooperation may involve visits from researchers of
both sides.

\subsection{Scientific equipment available (Apparative Ausstattung)}

In early 2003, the Genome Informatics research group aquired workstations and
servers from Sun Microsystems that were integrated in the CeBiTec
infrastructure provided by the Bioinformatics Resource Facility (BRF), where
they are available for use by any CeBiTec member. As a result, the members of
the Genome Informatics group have complete access to the BRF infrastructure,
including a cluster with 472 CPUs, a 10-Gigabit Ethernet network, a
44-Terabyte Backup server, and various compute servers with up to 96
GB of main storage. According to statement of the BRF director, the
existing infrastructure is sufficient for the requirements of the project
proposed here. Personal computers for the scientific staff and students are
available from the Genome Informatics research group resources.

\subsection{Your institution's general contribution (Laufende Mittel für
Sachausgaben)}

The Genome Informatics group has the necessary infrastructure of a scientific
working group with office rooms, office equipment and a secretary. The project
requested here will benefit from this infrastructure made available at a value,
roughly estimated, of 1,000 EUR annually (including office material and travel
expenses).

\subsection{Conflicts of interest with economic activities (Interessenkonflikte
bei wirtschaftlichen Aktivitäten)}

\enlargethispage{4.5ex}

None.

\subsection{Other requirements (Sonstige Voraussetzungen)}

None.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% ERKLÄRUNGEN
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Declaration (Erklärungen)}

A request for funding this project has not been submitted to any other
addressee. In the event that I submit such request, I will inform the Deutsche
Forschungsgemeinschaft immediately.

The DFG liaison officer of Bielefeld University, Herr Prof.\ Egelhaaf,
was informed about this application.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% UNTERSCHRIFT
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Signature (Unterschrift)}
\thispagestyle{empty}

\vspace{2ex}Bielefeld, 19. August 2008

\vspace{2ex}\hspace{17em}(Prof.\ Jens Stoye)

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% ANLAGEN
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{List of attachments (Verzeichnis der Anlagen)}

{\bf TODO S\'ergio/Z\'e: get letter from Felipe}

{\bf TODO Jens: get letter from Weisshaar}

\begin{enumerate}
\item Curriculum vitae of the applicant (DFG-Vordruck 10.04)
\item Complete publication list of the applicant
\item Letter of co-operation agreement from Dr.\ Felipe Rodrigues da Silva,
      Embrapa, Brazil
\item Letter of co-operation agreement from Prof.\ Bernd Weisshaar,
      Bielefeld University
\item CD-ROM with electronic versions of all request documents
\end{enumerate}

All attachments do not need to be returned.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% APPENDIX
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\appendix

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% LITERATUR
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{References (Literatur zum Antrag)}

\renewcommand{\refname}{\vspace*{-4ex}}
\bibliographystyle{abbrv}
\bibliography{biblio}

\end{document}
